### DISCUSSION TITLE
PrismaClient with hot-loading Next.js

### DISCUSSION THREAD

**Discussion by joaocasarin** [prisma/prisma#17641] [Q&A]
Well, as per [this Prisma docs](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#prevent-hot-reloading-from-creating-new-instances-of-prismaclient), I should use a single instance of `PrismaClient` and set it in the global scope.
However, after Node.js updated to v16, the `NodeJS.Global` doesn't exist anymore, showing this error:
![image](https://user-images.githubusercontent.com/48847394/215641332-3570600a-a4ac-4406-b15e-753a8085e54f.png)

With that, it's clear that the documentation is not actually following the updates of Node.js/TypeScript.

What should I do to proceed here?
This is my current code:
```ts
import { PrismaClient } from '@prisma/client';

interface CustomNodeJsGlobal extends NodeJS.Global {
    prisma: PrismaClient;
}

declare const global: CustomNodeJsGlobal;

const prisma = global.prisma || new PrismaClient();

if (process.env.NODE_ENV === 'development') global.prisma = prisma;

export default prisma;
```

**Comment by nurul3101** [comment]
Hello @joaocasarin 👋 

Can you try this code to instantiate a single PrismaClient instance?

```typescript
import { PrismaClient } from "@prisma/client";
declare let global: { prisma: PrismaClient };

let prisma: PrismaClient

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient()
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient()
  }
  prisma = global.prisma
}
export default prisma
```
Let me know if you encounter any issues.

**Reply by joaocasarin** [reply]
That works, indeed. Thanks!

But I don't know if that's supposed to fix another thing. I have a Next.js application deployed to Vercel, and I have a prisma client as default (can be seen [here](https://github.com/joaocasarin/tcc-frontend/blob/main/utils/database.ts)). That's a production environment, and that client seems to be similar to the production part of yours and yet, when I open multiple tabs of that website I get an error about Prisma connection. Should that be happening?

**Reply by janpio** [reply]
What error are you getting?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
